tools/respecDocWriter.js

```diff
@@ -91,6 +95,11 @@ async function fetchAndWrite(
     }
     await checkIfReSpec(page);
     const html = await generateHTML(page, url);
+    const abortOnWarning = whenToHalt.haltOnWarn && haltFlags.warn;
```

Comment:
I feel like the the `handleConsoleMessages()` should now move into the `generateHTML()` function, and that the `generateHTML()` should return the html AND the if it emitted an error and/or a warning.

Reply:
`handleConsoleMessages()` must be called before `page.goto()`, that was the point of #2057.

Reply:
Ah, keep forgetting. Sorry about that.